source/elements/oneDAL/source/common_interface/error_handling.rst

```diff
@@ -1,3 +1,127 @@
+classes are abstract. Leaf classes are also inherited from the corresponding STL
+exceptions. Only these classes can be thrown.
+
+The root of exception hierarchy is ``onedal::exception``. All throwing
+exceptions are inheritors of this one.
+
+::
+
+    class onedal::exception;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - onedal::exception
+     - Base class for all throwing exceptions.
+     - Yes
+
+
+``onedal::logic_error``, ``onedal::runtime_error`` and ``onedal::bad_alloc`` are inheritors of ``onedal::exception``.
+
+::
+
+    class onedal::logic_error : public onedal::exception;
+    class onedal::runtime_error : public onedal::exception;
+    class onedal::bad_alloc :  public onedal::exception, public std::bad_alloc;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - onedal::logic_error
+     - Reports violations of preconditions and invariants.
+     - Yes
+   * - onedal::runtime_error
+     - Reports violations of postconditions and other errors that are due to events beyond the scope of the program.
+     - Yes
+   * - onedal::bad_alloc
+     - Reports failure to allocate storage.
+     - No
+
+Precondition and invariant errors that are represented by ``onedal::logic_error``
+divide in subgroups.
+
+::
+
+    class onedal::invalid_argument :  public onedal::logic_error, public std::invalid_argument;
+    class onedal::domain_error :  public onedal::logic_error, public std::domain_error;
+    class onedal::out_of_range :  public onedal::logic_error, public std::out_of_range;
+    class onedal::unimplemented_error :  public onedal::logic_error, public std::logic_error;
+    class onedal::unavailable_error :  public onedal::logic_error, public std::logic_error;
+
+.. list-table::
+   :widths: 30 65 5
+   :header-rows: 1
+
+   * - Exception
+     - Description
+     - Abstract
+   * - onedal::invalid_argument
+     - Reports situations where the argument has
+       not been accepted.
+     - No
+   * - onedal::domain_error
+     - Reports situations where the argument is outside of the domain on which
+       an operation is defined. Higher priority than
+       ``onedal::invalid_argument``.
+     - No
+   * - onedal::out_of_range
+     - Reports situations where index is out of range. Higher priority
+       than ``onedal::invalid_argument``.
+     - No
+   * - onedal::unimplemented_error
+     - Reports errors that arise because an algorithm or method is not implemented.
+     - No
+   * - onedal::unavailable_error
+     - Reports situations when a device or data is not available.
+     - No
+
+If an error occurs during function execution after preconditions and invariants
+were checked, it is reported via ``ondeal::runtime_error`` inheritors. oneDAL distinguishes
+OS facilities and range errors in internal computations, while other errors are
```

Comment:
sounds confusing 
need a better wording for OS error, with this wording I can't immediately associate it and range errors